lib/irslackd.js

```diff
@@ -277,6 +278,11 @@ class Irslackd {
       }
     }
 
+    // Check for CTCP TYPING message
+    if (message.charCodeAt(0) === 1 && message.substr(1, 7) === 'TYPING ') {
```

Comment:
Replace this and the `ACTION` check above with a common `isCtcp(message, '...')` method. Might also need a `getCtcpParam(message, '...')` method to use with `ACTION` above.

Reply:
Wrap in a `ircUser.typingNotificationsEnabled()` conditional. (See `IrcUser::threadsEnabled()` for another user-preference example.) The feature can be off by default as it might confuse old clients.

Reply:
I have made these changes (added `isCtcpCommand` and `ircUser.typingNotificationsEnabled()`).  

I didn't think we needed a `getCtcpParam`. method, but I can add that if you want me to.